homeassistant/components/wevolor/cover.py

```diff
@@ -0,0 +1,96 @@
+"""Cover component for shades controlled by the Wevolor controller."""
+
+from __future__ import annotations
+
+from pywevolor import Wevolor
+
+from homeassistant.components.cover import (
+    SUPPORT_CLOSE,
+    SUPPORT_CLOSE_TILT,
+    SUPPORT_OPEN,
+    SUPPORT_OPEN_TILT,
+    SUPPORT_STOP,
+    SUPPORT_STOP_TILT,
+    CoverDeviceClass,
+    CoverEntity,
+)
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Wevolor shades."""
+
+    entities = []
+    wevolor = hass.data[DOMAIN][config_entry.entry_id]
+
+    channel_ids = range(1, config_entry.data["channel_count"] + 1)
+
+    for channel_id in channel_ids:
+        entity = WevolorShade(wevolor, channel_id, config_entry.data["support_tilt"])
+        entities.append(entity)
+
+    async_add_entities(entities, True)
+
+
+class WevolorShade(CoverEntity):
+    """Cover entity for control of Wevolor remote channel."""
+
+    _channel: int | None = None
```

Comment:
Cleanup, not needed, as set in the constructor.

```suggestion
    _attr_assumed_state = True
```